student-preferred-name-portlet-webapp/src/main/java/edu/wisc/portlet/preferred/service/PreferredNameServiceImpl.java

```diff
@@ -59,16 +59,20 @@ public String getStatus(PreferredName ldapPn) {
             && (jdbcPn == null
             || (StringUtils.isEmpty(jdbcPn.getFirstName()) && StringUtils
             .isEmpty(jdbcPn.getMiddleName()) && StringUtils.isEmpty(jdbcPn.getLastName()))
-        )
-            ) {
+            )){
             return "(not set)";
-        } else {
-
-            if (ldapPn.equals(jdbcPn)) {
+        }else if(jdbcPn == null || 
+                (StringUtils.isEmpty(jdbcPn.getFirstName()) && 
+                 StringUtils.isEmpty(jdbcPn.getMiddleName()) && 
+                 StringUtils.isEmpty(jdbcPn.getLastName()))){ 
```

Comment:
Looks like a `PreferredName` can represent a preferred name deletion, and than it does so by the first, middle, and last name fields being empty.  Could refactor this logic into a `isRepresentingDeletion()` property on the `PreferredName` object, which would be eminently unit testable and might feel more object oriented.